### PR TITLE
Remove deprecated props

### PR DESCRIPTION
**Pull request by adguernier**
## Problem

Some props are deprecated

## Solution

Remove them

## TODO
Remove: 
- [x] `disableNotification` from `useCheckAuth`
- [x] `disableNotification` from `useLogoutIfAccessDenied`
- [x] ~`resource` from `useDeleteWithConfirmController`~
- [x] ~`resource` from `useDeleteWithUndoController`~
- [x] `data` from `CreateControllerResult`
- [x] `referenceSource` from `useReferenceInputController`
- [ ] `saving` from `SaveContextValue` ? :thinking: -> later
- [ ] `useGetValidationErrorMessage` ? :thinking:  -> later
	- [ ] `<ArrayInput>` has to be adapted 
- [x] ~`useLocale`~ 
- [x] ~`useSetLocale`~
- [x] `linkToRecord`
- [x] `resolveRedirectTo` 
- [x] `onError` type from ra-core type.ts
- [x] `formClassName` from `FieldProps`
- [ ] `fullWidth` from `FieldProps`
	- [ ] many occurences
- [x] `PublicFieldProps` from `FieldProps`
- [x] `InjectedFieldProps` from `FieldProps`
- [x] `formClassName` from `CommonInputProps`
- [ ] `sidebarIsOpen` from `DashboardMenuItem` -> later
	- [ ] check the impact in ra-no-code before
- [ ] `sidebarIsOpen` from `MenuItemLink`  -> later 
 	- [ ] check the impact in ra-no-code before
- [x] ~`rowStyle` from `Datagrid`~
- [x] `PaginationLimit` 
- [x] ~`rowStyle` from `SimpleList`~
- [x] `theme` from `ThemeProvider`

### CODE REVIEW

File: docs/Upgrade.md [hunk 1]
```diff
@@ -924,6 +924,35 @@ The `<InputHelperText>` component no longer accepts a `touched` prop. This prop
 
 If you were using this prop, you can safely remove it.
 
+## `disableNotification` Param of `useCheckAuth` Was Removed
+
+The `useCheckAuth` hook no longer accepts the deprecated `disableNotification` param. To disabe the notification at the `checkAuth` call, `authProvider.checkAuth()` should return a rejected promise with a `falsy` message:
```
Comment (fzaninotto) [inline docs/Upgrade.md:929]:
```suggestion
The `useCheckAuth` hook no longer accepts the deprecated `disableNotification` param. To disable the "Authentication required" notification when calling `checkAuth`, `authProvider.checkAuth()` should return a rejected promise with the value `{ message: false }`:
```

File: docs/Upgrade.md [hunk 2]
```diff
@@ -924,6 +924,35 @@ The `<InputHelperText>` component no longer accepts a `touched` prop. This prop
...
+```ts
+const authProvider: AuthProvider = {
+    //...
+    checkAuth: () => Promise.reject({ message: false }),
+}
+```
+
+## `disableNotification` Param of `useLogoutIfAccessDenied` Was Removed
+
+The `useLogoutIfAccessDenied` hook no longer accepts the deprecated `disableNotification` param. To disabe the notification at the `checkError` call, `authProvider.checkError()` should return a rejected promise with a `falsy` message:
```
Comment (fzaninotto) [inline docs/Upgrade.md:940]:
```suggestion
The `useLogoutIfAccessDenied` hook no longer accepts the deprecated `disableNotification` param. To disable the "Authentication required" notification when `checkError` is called, `authProvider.checkError()` should return a rejected promise with the value `{ message: false }`:
```

File: docs/Upgrade.md [hunk 3]
```diff
@@ -924,6 +924,35 @@ The `<InputHelperText>` component no longer accepts a `touched` prop. This prop
 
 If you were using this prop, you can safely remove it.
 
+## `disableNotification` Param of `useCheckAuth` Was Removed
```
Comment (fzaninotto) [inline docs/Upgrade.md:927]:
```suggestion
## `useCheckAuth` No Longer Accepts A `disableNotification` Param
```

File: docs/Upgrade.md [hunk 4]
```diff
@@ -924,6 +924,35 @@ The `<InputHelperText>` component no longer accepts a `touched` prop. This prop
...
+The `useCheckAuth` hook no longer accepts the deprecated `disableNotification` param. To disabe the notification at the `checkAuth` call, `authProvider.checkAuth()` should return a rejected promise with a `falsy` message:
+
+```ts
+const authProvider: AuthProvider = {
+    //...
+    checkAuth: () => Promise.reject({ message: false }),
+}
+```
+
+## `disableNotification` Param of `useLogoutIfAccessDenied` Was Removed
```
Comment (fzaninotto) [inline docs/Upgrade.md:938]:
```suggestion
## `useLogoutIfAccessDenied` No Longer Accepts A `disableNotification` Param
```

**Review by fzaninotto — COMMENTED** [review]
So far, so good!

**Review by fzaninotto — APPROVED** [review]

**Comment by fzaninotto** [comment]
I think we can de-deprecate and keep in v5 the following hooks

- `useLocale`
- `useSetLocale`

**Review by djhi — APPROVED** [review]

**Review by erwanMarmelab — APPROVED** [review]

**Review by djhi — APPROVED** [review]

**Pull request merged**